dask/array/tests/test_reductions.py

```diff
@@ -287,3 +255,30 @@ def test_0d_array():
 def test_reduction_on_scalar():
     x = da.from_array(np.array(1.0), chunks=())
     assert (x == x).all()
+
+
+def test_tree_reduce_depth():
+    x = da.from_array(np.arange(242).reshape((11, 22)), chunks=(3, 4))
+    # Check that tree depth is 2
+    o = x.sum(axis=0, max_leaves=3)
+    d1 = o.dask[(o.name, 0)]
+    assert len(d1[1]) == 2
+    d2 = o.dask[d1[1][0]]
+    assert len(d2[1]) == 3
+    assert all(i[0].startswith('atop') for i in d2[1])
+    # Check that tree depth is 3
+    o = x.sum(axis=1, max_leaves=2)
+    d1 = o.dask[(o.name, 0)]
+    assert len(d1[1]) == 2
+    d2 = o.dask[d1[1][0]]
+    assert len(d2[1]) == 2
+    d3 = o.dask[d2[1][0]]
+    assert len(d3[1]) == 2
+    assert all(i[0].startswith('atop') for i in d3[1])
```

Comment:
You could use `dask.core.get_deps` to get the `dependencies` dict and then use this dictionary in more sophisticated tests like the following:

```python
dependencies, dependents = get_deps(x.sum(max_leaves={0: 2, 1: 3}.dask)
assert max(map(len, dependencies.values()) == 2 * 3
```

There are presumably several such interesting configurations that would be useful to verify now, rather than several months from now when someone screws with this code unknowingly.